operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/AbstractWorkflowExecutor.java

```diff
@@ -101,9 +102,15 @@ protected synchronized void handleNodeExecutionFinish(
 
   protected <R> boolean isConditionMet(Optional<Condition<R, P>> condition,
       DependentResource<R, P> dependentResource) {
+    if (condition.isEmpty()) {
+      return true;
+    }
+    var resources = dependentResource instanceof BulkDependentResource
+        ? ((BulkDependentResource) dependentResource).getSecondaryResources(primary, context)
+        : dependentResource.getSecondaryResource(primary, context).orElse(null);
+
     return condition.map(c -> c.isMet(primary,
-        dependentResource.getSecondaryResource(primary, context).orElse(null),
-        context))
+        (R) resources, context))
```

Comment:
This shouldn't work because `R` can never be a `Map` in a `DependentResource` declaration so this should most likely cause a `ClassCastException`.

Reply:
That thing does not get into the bytecode, but also there is a test for this in the PR. Will try to make it more nice though.

Reply:
Checked but this is still the simplest way to do locally, this will change probably within this issue:
https://github.com/java-operator-sdk/java-operator-sdk/issues/1689

Reply:
will improve this sample so it shows more the issue with the Bulk resources.